source/Nuke.Common/Tools/MSBuild/MSBuildSettings.cs

```diff
@@ -44,5 +46,130 @@ private string GetToolPath()
         {
             return MSBuildToolPathResolver.Resolve(MSBuildVersion, MSBuildPlatform);
         }
+
+
+        internal string GetFileLogger()
+        {
+            if (FileLogger == null)
+                return null;
+
+            var result = new StringBuilder();
+
+            // A maximum of 9 loggers are supported.
+            for (var i = 0; i < FileLogger.Count && i < 9; i++)
+            {
+                if (i != 0)
+                    result.Append(' ');
+
+                var fileLogger = FileLogger[i];
+                var builder = new MSBuildParameterBuilder($"/flp{i + 1}", true)
+                   .Add("PerformanceSummary", fileLogger.PerformanceSummary)
+                   .Add("Summary", "NoSummary", fileLogger.Summary)
+                   .Add("ErrorsOnly", fileLogger.ErrorsOnly)
+                   .Add("WarningsOnly", fileLogger.WarningsOnly)
+                   .Add("NoItemAndPropertyList", fileLogger.NoItemAndPropertyList)
+                   .Add("ShowCommandLine", fileLogger.ShowCommandLine)
+                   .Add("ShowTimestamp", fileLogger.ShowTimestamp)
+                   .Add("ShowEventId", fileLogger.ShowEventId)
+                   .Add("EnableMPLogging", "DisableMPLogging", fileLogger.MultiProcessorLogging)
+                   .Add("Verbosity", fileLogger.Verbosity?.ToString())
+                   .Add("LogFile", fileLogger.LogFile)
+                   .Add("Append", fileLogger.Append)
+                   .Add("Encoding", fileLogger.Encoding);
+
+                result.Append(builder.Render());
+            }
+
+            return result.ToString();
+        }
+
+        internal string GetConsoleLogger()
+        {
+            if (ConsoleLogger == null)
+                return null;
+
+            var builder = new MSBuildParameterBuilder("/consoleLoggerParameters", emptyParametersAllowed: false)
```

Comment:
This is exactly the part I want the code generation to take over

Reply:
Yeah, I agree that that would be nice. Question is if this case is common enough to warrant the effort to write code generation for it?   If it is, then do you have any ideas for how the specifications would best be extended to support this?

Reply:
If you don't have any initial input, I'll sit down and try to sketch up a suggestion about how the json schema could look to support this, and get back to you to see what you think.